### PR TITLE
Change generate-class-name for fonts, colors, assets

### PR DESCRIPTION
**Pull request by Cu-Toof**
Change the class name that will be generated.
Update pubspec.yaml file:

```
flutter_gen:
  output: lib/gen/

  assets:
    outputs:
      class_name: AppAssets //default is Assets

  fonts:
    outputs:
      class_name: AppFontFamily //default is FontFamily

  colors:
    inputs:
      - assets/colors/colors.xml 
    outputs:
      class_name: AppColor //default is ColorName
```

### CODE REVIEW

File: packages/core/lib/generators/colors_generator.dart [hunk 1]
```diff
@@ -15,25 +15,26 @@ import 'generator_helper.dart';
...
+  FlutterGenColors genColors,
 ) {
-  if (colors.inputs.isEmpty) {
+  if (genColors.inputs.isEmpty) {
     throw const InvalidSettingsException(
         'The value of "flutter_gen/colors:" is incorrect.');
   }
 
   final buffer = StringBuffer();
+  final className = genColors.outputs?.className ?? 'FontFamily';
```
Comment (thangnc) [inline packages/core/lib/generators/colors_generator.dart:26]: Why did you use `FontFamily` in color generator?

File: packages/core/lib/flutter_generator.dart [hunk 2]
```diff
@@ -56,8 +56,11 @@ class FlutterGenerator {
     }
 
     if (flutterGen.colors.enabled && flutterGen.colors.inputs.isNotEmpty) {
-      final generated =
-          generateColors(pubspecFile, formatter, flutterGen.colors);
+      final generated = generateColors(
+        pubspecFile,
+        formatter,
+        flutterGen.colors,
+      );
```
Comment (thangnc) [inline packages/core/lib/flutter_generator.dart:63]: I dont think you need to update this call formatter?

**Review by thangnc — COMMENTED** [review]
Btw, you need to fix unit test @Cu-Toof

**Comment by thangnc** [comment]
It seems to be duplicated with this PR https://github.com/FlutterGen/flutter_gen/pull/279

File: packages/core/lib/flutter_generator.dart (hunk 2 above)
Comment (Cu-Toof) [reply packages/core/lib/flutter_generator.dart]: I think so, I have updated this code.

File: packages/core/lib/generators/colors_generator.dart (hunk 1 above)
Comment (Cu-Toof) [reply packages/core/lib/generators/colors_generator.dart]: My mistake, I have fixed it.

**Comment by Cu-Toof** [comment]
> 

@thangnc I have reviewed this PR [#279](https://github.com/FlutterGen/flutter_gen/pull/279).
I think we can custom more than one output's information. But this pr is implementing `output_class` is just string type.
Replace it, I implement `output` as an object, code is scalable with other output's customize like file name, prefix name...

File: packages/core/lib/generators/assets_generator.dart [hunk 3]
```diff
@@ -306,27 +307,30 @@ String _flatStyleDefinition(
...
       .whereType<_Statement>()
       .toList();
-  return _flatStyleAssetsClassDefinition(statements);
+  final className = config.flutterGen.assets.outputs?.className;
+  return _flatStyleAssetsClassDefinition(className, statements);
 }
 
-String _flatStyleAssetsClassDefinition(List<_Statement> statements) {
+String _flatStyleAssetsClassDefinition(
+    String? className, List<_Statement> statements) {
```
Comment (thangnc) [inline packages/core/lib/generators/assets_generator.dart:315]: Please follow the current code convention. Please refer https://github.com/FlutterGen/flutter_gen/blob/main/packages/core/lib/generators/assets_generator.dart#L289

File: packages/core/lib/settings/config.dart [hunk 4]
```diff
@@ -47,13 +47,19 @@ flutter_gen:
     enabled: true
     package_parameter_enabled: false
     style: dot-delimiter
+    outputs:
+      class_name: Assets
```
Comment (thangnc) [inline packages/core/lib/settings/config.dart:51]: Becoz you checked in this line [ in Asset](https://github.com/FlutterGen/flutter_gen/pull/280/files#diff-8dc8f78c31449fdef325c4ed0e8da50067716f50c6370fc9e477e3627219e63fR332), so the default config no need `outputs.class_name` param, right?

File: packages/core/lib/settings/config.dart [hunk 5]
```diff
@@ -47,13 +47,19 @@ flutter_gen:
...
   fonts:
     enabled: true
+    outputs:
+      class_name: FontFamily
 
   colors:
     enabled: true
     inputs: []
+    outputs:
+      class_name: ColorName
```
Comment (thangnc) [inline packages/core/lib/settings/config.dart:62]: Same above comment becoz you checked in this line https://github.com/FlutterGen/flutter_gen/pull/280/files#diff-4f1088f5086a316577b245788f11453e91629f92b823d1fa58936750f6f1f518R26

File: packages/core/lib/settings/config.dart [hunk 6]
```diff
@@ -47,13 +47,19 @@ flutter_gen:
     enabled: true
     package_parameter_enabled: false
     style: dot-delimiter
+    outputs:
+      class_name: Assets
     
   fonts:
     enabled: true
+    outputs:
+      class_name: FontFamily
```
Comment (thangnc) [inline packages/core/lib/settings/config.dart:56]: Same as above comment becoz you checked here https://github.com/FlutterGen/flutter_gen/pull/280/files#diff-74b9255cc2b20267c715b8162a5fcd4e4d55b3b0fb333a5bc23b52deb5d207e4R22

**Review by thangnc — COMMENTED** [review]
@Cu-Toof please fix my PR's comments. Btw, you did not add unit test?

**Comment by wasabeef** [comment]
I will merge this PR after all fix.

@thangnc I appreciate your help.

**Pull request closed without merging**